libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -410,8 +410,9 @@ void AP_UAVCAN::do_cyclic(void)
                             repeat_send = false;
                             uavcan::equipment::actuator::ArrayCommand msg;
 
-                            uint8_t k = 0;
-                            for (uint8_t i = 0; i < UAVCAN_RCO_NUMBER && k < 15; i++) {
+                            uint8_t i;
+                            // UAVCAN can hold maximum of 15 commands in one frame
+                            for (i = 0; starting_servo < UAVCAN_RCO_NUMBER && i < 15; i++, starting_servo++) {
```

Comment:
I'm tired so I might be wrong (again), but shouldn't the `i++` be in the same place that `k++` was? Otherwise `i` will increase even when a command wasn't added.